core/src/main/scala/org/apache/spark/storage/StorageUtils.scala

```diff
@@ -118,8 +140,20 @@ class StorageStatus(val blockManagerId: BlockManagerId, val maxMem: Long) {
         } else {
           None
         }
+      case BroadcastBlockId(broadcastId, _) =>
+        // Actually remove the block, if it exists
+        if (_broadcastBlocks.contains(broadcastId)) {
+          val removed = _broadcastBlocks(broadcastId).remove(blockId)
+          // If the given RDD has no more blocks left, remove the RDD
```

Comment:
actually now I am thinking there is a lot of copy-pasting that could be cleaned up.  you could mostly merge this w/ the block above if you did something like:

```
case rddOrBlockId @ (_: BroadcastBlockId | _: RDDBlockId) =>
  val (id, blockMap) = getIdAndBlockMap(rddOrBlockId)
  val removed = blockMap(id).remove(blockId)
  ...
```

where the helper function `getIdAndBlockMap` is something like:

```
def getIdAndBlockMap(blockId: BlockId): (Int, Map[BlockId, BlockStatus]) = blockId match {
  case RDDBlockId(rddId, _) => (rddId, _rddBlocks)
  case BroadcastBlockId(broadcastId, _) => (broadcastId, _broadcastBlocks)
}
```

and then you could do a similar thing in a few other places.  You could also take this a step further, and even merge `_rddBlocks` and `_broadcastBlocks` into a `EnumMap[BlockType, Map[BlockId, BlockStatus]]` if you made a new `public enum BlockType{ RDD,Broadcast}`, but that might not really help much since at the end of the day you do want separate getter methods for the RDD and Broadcast stuff